jsonnet/dashboards/apps/openstack.libsonnet

```diff
@@ -0,0 +1,1560 @@
+            legendFormat='{{instance}}-routers-inactive'
+          )
+        );
+
+      local neutronPortsPanel =
+        graphPanel.new(
+          title='Ports',
+          datasource='$datasource',
+          decimals=0,
+          min=0,
+        )
+
+        .addTarget(
+          prometheus.target(
+            'count by (instance) (openstack_neutron_port{job=~"$job",instance=~"$instance",cluster=~"$cluster"})',
+            legendFormat='{{instance}}-ports-total'
+          )
+        )
+
+        .addTarget(
+          prometheus.target(
+            'count by (instance)(openstack_neutron_port{status!="ACTIVE", job=~"$job",instance=~"$instance",cluster=~"$cluster"})',
+            legendFormat='{{instance}}-ports-inactive'
+          )
+        )
+
+        .addTarget(
+          prometheus.target(
+            'openstack_neutron_ports_no_ips{job=~"$job",instance=~"$instance",cluster=~"$cluster"}',
+            legendFormat='{{instance}}-ports-no-IPs'
+          )
+        );
+
+      local neutronFloatingIPsPanel =
+        graphPanel.new(
+          title='Floating IPs',
+          datasource='$datasource',
+          decimals=0,
+          min=0,
+        )
+
+        .addTarget(
+          prometheus.target(
+            'openstack_neutron_floating_ips{job=~"$job",instance=~"$instance",cluster=~"$cluster"}',
+            legendFormat='{{instance}}-IP-total'
+          )
+        )
+
+        .addTarget(
+          prometheus.target(
+            'openstack_neutron_floating_ips_associated_not_active{job=~"$job",instance=~"$instance",cluster=~"$cluster"}',
+            legendFormat='{{instance}}-IPs-associated-inactive'
+          )
+        );
+
+      local neutronIPsUsedBySubnetsPanel =
+        graphPanel.new(
+          title='IPs used by subnets',
+          datasource='$datasource',
+          description='The usage of available IP addresses broken down by subnets',
+          decimals=1,
+          min=0,
+          max=1,
+          format='percentunit',
+        )
+
+        .addTarget(
+          prometheus.target(
+            'sum by (job, instance, ip_version, subnet_name) (openstack_neutron_network_ip_availabilities_used{job=~"$job",instance=~"$instance",cluster=~"$cluster"}) / sum by (job, instance, ip_version, subnet_name)(openstack_neutron_network_ip_availabilities_total{job=~"$job",instance=~"$instance",cluster=~"$cluster"})',
+            legendFormat='{{instance}}-{{subnet_name}}'
+          )
+        );
+
+      local securityGroupsPanel =
+        graphPanel.new(
+          title='Security groups',
+          datasource='$datasource',
+          description='The number of Security groups managed by Neutron',
+          decimals=0,
+        )
+
+        .addTarget(
+          prometheus.target(
+            'openstack_neutron_security_groups{job=~"$job",instance=~"$instance",cluster=~"$cluster"}',
+            legendFormat='{{instance}}-sec-groups'
+          )
+        );
+
+      local routerDetailsPanel =
```

Comment:
Probably you can do the same for `*DetailsPanel` as for the status panels, as I also count at least 3-4 panels with the exact same logic.